ibm/service/cis/resource_ibm_cis_mtls_app.go

```diff
@@ -0,0 +1,285 @@
+	zoneID, _, _ := flex.ConvertTftoCisTwoVar(d.Get(cisDomainID).(string))
+
+	sess.Crn = core.StringPtr(crn)
+	OptionsApp := sess.NewCreateAccessApplicationOptions(zoneID)
+
+	if app_val, ok := d.GetOk(cisMtlsAppName); ok {
+		OptionsApp.SetName(app_val.(string))
+	}
+
+	if host_val, ok := d.GetOk(cisMtlsHostName); ok {
+		OptionsApp.SetDomain(host_val.(string))
+	}
+
+	if dur_val, ok := d.GetOk(cisMtlsDuration); ok {
+		OptionsApp.SetSessionDuration(dur_val.(string))
+	}
+
+	resultApp, responseApp, operationErrApp := sess.CreateAccessApplication(OptionsApp)
+
+	if operationErrApp != nil || resultApp == nil {
+		return fmt.Errorf("[ERROR] Error creating access application  %v %v %v", operationErrApp, resultApp, responseApp)
+	}
+
+	d.SetId(flex.ConvertCisToTfThreeVar(*resultApp.Result.ID, zoneID, crn))
+
+	//save appID
+	appId := *resultApp.Result.ID
+
+	// Create an access policy
+	policyRuleModel := &mtlsv1.PolicyRulePolicyCertRule{
+		Certificate: map[string]interface{}{"certifcate": "CA root certificate"},
+	}
+	policyCnModel := &mtlsv1.PolicyCnRuleCommonName{
+		CommonName: core.StringPtr("Access Testing CA"),
+	}
+	policyModel := &mtlsv1.PolicyRulePolicyCnRule{
+		CommonName: policyCnModel,
+	}
+	optionsPolicy := sess.NewCreateAccessPolicyOptions(zoneID, appId)
+
+	// get policy name and action/decsion
+	if policy_val, ok := d.GetOk(cisMtlsPolicyName); ok {
+		optionsPolicy.SetName(policy_val.(string))
+	}
+	if action_val, ok := d.GetOk(cisMtlsPolicyAction); ok {
+		optionsPolicy.SetDecision(action_val.(string))
+	}
+
+	optionsPolicy.SetInclude([]mtlsv1.PolicyRuleIntf{policyModel, policyRuleModel})
+	resultPolicy, responsePolicy, operationErrPolicy := sess.CreateAccessPolicy(optionsPolicy)
+
+	if operationErrPolicy != nil || resultPolicy == nil {
+		return fmt.Errorf("[ERROR] Error creating app policy  %v", responsePolicy)
+	}
+
+	d.SetId(flex.ConvertCisToTfThreeVar(*resultApp.Result.ID, zoneID, *resultPolicy.Result.ID))
+	return ResourceIBMCISMtlsAppRead(d, meta)
+
+}
+func ResourceIBMCISMtlsAppRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess)
+	}
+
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+	appID, zoneID, policyID, _ := flex.ConvertTfToCisThreeVar(d.Id())
+	getAppOptions := sess.NewGetAccessApplicationOptions(zoneID, appID)
+	getAppResult, getAppResp, getAppErr := sess.GetAccessApplication(getAppOptions)
+
+	if getAppErr != nil || getAppResult == nil {
+		return fmt.Errorf("[ERROR] Error getting app deatil  %v", getAppResp)
+	}
+
+	getPolicyOptions := sess.NewGetAccessPolicyOptions(zoneID, appID, policyID)
+	getPolicyResult, getPolicyResp, getPolicyErr := sess.GetAccessPolicy(getPolicyOptions)
+
+	if getPolicyErr != nil || getPolicyResult == nil {
+		return fmt.Errorf("[ERROR] Error getting Policy  detail  %v", getPolicyResp)
+	}
```

Comment:
Set back all fields defined in schem (d.Set) in read ..We have just made API calls